app.js

```diff
@@ -45,18 +49,28 @@ function displayPhrases() {
 
     for (const phrase of character.phrases) {
         // > create an <li> element
+        const li = document.createElement('li');
         // > set the text of the li to the phrase
+        li.textContent = phrase;
         // > append the <li> to the phrases list
+        phraseList.append(li);
     }
 }
 
 addButton.addEventListener('click', () => {
-    // > Get the phrase from the value of the phrase input
-    // > Use ".push" to add the phrase to the end of the character phrases array
-    // > Re-display the phrases
-
-    // reset the input
-    phraseInput.value = '';
+    // > Add conditional logic that prevents the phrase from being added is the value is empty
+    if (phraseInput.value === '') {
+        return '';
+    } else {
+        // > Get the phrase from the value of the phrase input
+        const phrase = phraseInput.value;
+        // > Use ".push" to add the phrase to the end of the character phrases array
+        character.phrases.push(phrase);
+        // > Re-display the phrases
+        displayPhrases();
+        // reset the input
+        phraseInput.value = '';
+    }
```

Comment:
check out this move:
```
if (!phraseInput.value) {
    return;
} else {
    // code you used to push phrase and display
}
```
this is a little advanced, but in JavaScript an empty string (and the number 0 , null , NaN , a boolean false , and undefined variables) are “falsy” so if you run ```if (falsy thing) {this code won't run because falsy}```.
AND THEN, since you DO want to run the code in the ```if``` check in this situation, the ```!``` operator before something makes it the opposite — ```!true``` is false and ```!false``` is true. So ```!phraseInput.value``` is true when phraseInput.value is an empty string.
And you don't need to ```return``` and empty string, you can simply ```return``` and it'll exit the ```if``` check without finishing the rest of the code.

but the way you did it is totally fine! hopefully that wasn't too confusing or overwhelming.